restdocs-openapi-gradle-plugin/src/test/kotlin/com/epages/restdocs/openapi/gradle/OpenApi20GeneratorTest.kt

```diff
@@ -26,6 +28,55 @@ class OpenApi20GeneratorTest {
         val openapi = OpenApi20Generator.generate(api)
 
         println(Json.pretty().writeValueAsString(openapi))
+        thenGetProductWith200ResponseIsGenerated(openapi, api)
+    }
+
+
+    @Test
+    fun `should convert multiple resource models to openapi`() {
+        val api = givenResourceModelsWithDifferentResponsesForSameRequest()
+
+        val openapi = OpenApi20Generator.generate(api)
+
+        println(Json.pretty().writeValueAsString(openapi))
+        thenGetProductWith200ResponseIsGenerated(openapi, api)
+        thenGetProductWith400ResponseIsGenerated(openapi, api)
+        thenDeleteProductIsGenerated(openapi, api)
+    }
+
+    private fun thenGetProductWith200ResponseIsGenerated(openapi: Swagger, api: List<ResourceModel>) {
+        then(openapi.basePath).isEqualTo("/api")
+        then(openapi.paths.getValue(api.get(0).request.path)).isNotNull
+        then(openapi.paths.getValue(api.get(0).request.path).get.consumes).contains(api.get(0).request.contentType)
+        then(openapi.paths.getValue(api.get(0).request.path).get.responses.get(api.get(0).response.status.toString())).isNotNull
+        then(openapi.paths.getValue(api.get(0).request.path).get.security.get(0).get("OAUTH2"))
+                .isEqualTo(api.get(0).request.securityRequirements!!.requiredScopes)
+        then(openapi.paths.getValue(api.get(0).request.path).get.responses.get(api.get(0).response.status.toString())!!
+                .examples.get(api.get(0).response.contentType)).isEqualTo(api.get(0).response.example)
+        then(openapi.paths.getValue(api.get(0).request.path).get.parameters.get(0).name)
+                .isEqualTo(api.get(0).request.pathParameters.get(0).name)
+        then(openapi.paths.getValue(api.get(0).request.path).get.parameters.get(1).name)
+                .isEqualTo(api.get(0).request.requestParameters.get(0).name)
```

Comment:
Please assign objects like `api.get(0)` or `openapi.paths.getValue(api.get(0).request.path).get.parameters.get(1)` to named variables for better readability and making it easier to identify the objects